### PR TITLE
feat: add a table select to choose a table as prompt

### PR DESCRIPTION
**Pull request by CorrectRoadH**
The PR is ready for review🥳

👀 for resolving the https://github.com/sqlchat/sqlchat/issues/51. add a picker to choose a table that the user wants to use is a simple solution.

But there are some problems that need to resolve🧑‍💻
- #54
- #58 
- #65
- #71
## What does the PR do:
- add loading status of fetch database schema after chosen database
- only send the selected table struct as prompt
- send all table struct as prompt when "All tables" is selected

## Screenshot
CleanShot 2023-05-02 at 15 54 33

## When will the data be changed👀
In the PR, I only add a new state called tableName in ConversationStore. and change the time of loading database schema.

the value `undefined` or "" of tableName denotes "All Tables", otherwise means a table.  

to update the selected table name
- table name will reset(default value mean "All Tables") when the connection is changed
- table select be change

to load schema
- table be selected 
- change connection

## What will happen if chat or select another table when the database schema is loading yet
Sometimes we meet some databases with very many tables. the loading will be long👀(although it has already been optimized in #65, the time may be up to a few seconds. So two requests may append yet). 

Nothing bug will happen.🥳 If the two requests to fetch two different schemas, the two schemas will be loaded completely! (happened in to select database when current is loading), If the request fetches two same schemas(happened in sending a message when the schema is loading ), the chat will wait and uses the schema from the request that is requested from the chat, instead of the request from selecting a table.👀 

## What time tableName is `undefined`

```typescript
<Select
  className="w-full px-4 py-3 !text-base"
  value={
    useConversationStore.getState().getConversationById(
      useConversationStore.getState().currentConversationId
    )?.tableName || ""
  }
```
When we changed to a database that didn't have any conversation, the tableName of conversation is `undefined`. So I need to add `||` to process the case.

## What is next?
a picker of the table is a temporary solution for resolving #51. I think it is not good enough. When we want to generate statements from multiple tables. the select can't do it. I think a selection that can select multiple tables may be a good solution 🤔

## Other
🤔I think the list of selected states of the table, rather than "All Tables" or a table name, is a better design. But I didn't have any good UI design ideas for it. So I use Select and a select table name variable yet.

### CODE REVIEW

File: src/components/ConnectionSidebar.tsx [hunk 1]
```diff
@@ -49,12 +59,41 @@ const ConnectionSidebar = () => {
...
+            });
+          }
         });
     } else {
       setIsRequestingDatabase(false);
     }
   }, [currentConnectionCtx?.connection]);
 
+  useEffect(() => {
+    const newTable =
```
Comment (tianzhou) [inline src/components/ConnectionSidebar.tsx:80]: tableList

**Review by tianzhou — CHANGES_REQUESTED** [review]

**Comment by CorrectRoadH** [comment]
@tianzhou PTAL

File: src/components/ConnectionSidebar.tsx [hunk 2]
```diff
@@ -126,6 +173,35 @@ const ConnectionSidebar = () => {
...
+                          conversationStore.currentConversationId
+                        )?.tableName || ""
+                      }
+                      itemList={tableList.map((table) => {
+                        return {
+                          label: table.name,
+                          value: table.name === t("connection.all-tables") ? "" : table.name
+                        };
+                      })}
+                      onValueChange={(databaseName) =>
```
Comment (tianzhou) [inline src/components/ConnectionSidebar.tsx:197]: tableName

File: src/components/ConnectionSidebar.tsx [hunk 3]
```diff
@@ -49,12 +59,40 @@ const ConnectionSidebar = () => {
...
+    const tableList =
+      connectionStore.databaseList.find(
+        (database) =>
+          database.connectionId === currentConnectionCtx?.connection.id &&
+          database.name === currentConnectionCtx?.database?.name
+      )?.tableList || [];
+
+    updateTableList([
+      {
+        name: t("connection.all-tables"),
```
Comment (tianzhou) [inline src/components/ConnectionSidebar.tsx:89]: To make it less error prone.

this should be name: ""

and in the the rendering part if name == "" then display t("connection.all-tables")

**Review by tianzhou — CHANGES_REQUESTED** [review]

**Comment by CorrectRoadH** [comment]
@tianzhou  PTAL again🤩

**Review by tianzhou — APPROVED** [review]
LGTM

**Pull request merged**